File: notes/src/main/java/com/abhijith/notes/util/ColorUtils.java
package com.abhijith.notes.util;

import android.graphics.Color;

public class ColorUtils {

    public static int getContrastColor(int backgroundColor) {
        double brightness = ColorUtils.calculateBrightness(backgroundColor);
        int textColor = (brightness < 128) ? Color.WHITE : Color.BLACK;
        return textColor;
    }

    public static double calculateBrightness(int color) {
        int red = Color.red(color);
        int green = Color.green(color);
        int blue = Color.blue(color);
        return (0.299 * red + 0.587 * green + 0.114 * blue);
    }
}

File: notes_ui/src/main/java/com/abhijith/notes/util/ColorUtils.kt
package com.abhijith.notes.util

import android.graphics.Color

object ColorUtils {
    fun getContrastColor(backgroundColor: Int): Int {
        val brightness =
            calculateBrightness(backgroundColor)
        return if (brightness < 128) Color.WHITE else Color.BLACK
    }

    private fun calculateBrightness(color: Int): Double {
        val red = Color.red(color)
        val green = Color.green(color)
        val blue = Color.blue(color)
        return 0.299 * red + 0.587 * green + 0.114 * blue
    }
}